src/features/goblins/rocket/Rocket.tsx

```diff
@@ -0,0 +1,387 @@
+import React, { useEffect, useState, useContext } from "react";
+import { useActor } from "@xstate/react";
+import { metamask } from "lib/blockchain/metamask";
+import { Panel } from "components/ui/Panel";
+import { Modal } from "react-bootstrap";
+import { ItemsModal } from "./ItemsModal";
+import { Button } from "components/ui/Button";
+import { Minting } from "features/game/components/Minting";
+import { ErrorMessage } from "features/auth/ErrorMessage";
+
+import brokenRocket from "assets/mom/mom_broken_rocket.gif";
+import fixedRocket from "assets/mom/mom_fixed_rocket.png";
+import launchingRocket from "assets/mom/mom_launching_rocket.gif";
+import burnMark from "assets/mom/mom_burnt_ground.png";
+import close from "assets/icons/close.png";
+import observatory from "assets/nfts/mom/observatory.gif";
+import { melonDuskAudio, rocketLaunchAudio } from "lib/utils/sfx";
+import momNpc from "assets/mom/mom_npc.gif";
+import scaffoldingLeft from "assets/mom/scaffolding_left.png";
+import scaffoldingRight from "assets/mom/scaffolding_right.png";
+import support from "assets/mom/launch-pad-material-2.png";
+import platform from "assets/mom/launch-pad-material-3.png";
+import woodPile from "assets/mom/launch-pad-material-4.png";
+import goblinHammering from "assets/mom/goblin_mechanic_1.gif";
+import goblinWelding from "assets/mom/goblin_mechanic_2.gif";
+import goblinForeman from "assets/mom/goblin_mechanic_3.gif";
+import metalSheetsPileFew from "assets/mom/metal-sheets-pile-few.png";
+import metalSheetsPileMany from "assets/mom/metal-sheets-pile-many.png";
+
+import { GRID_WIDTH_PX } from "features/game/lib/constants";
+import { Context } from "features/game/GoblinProvider";
+
+const ROCKET_LAUNCH_TO_DIALOG_TIMEOUT = 4000;
+const MELON_DUSK_SEEN = "isMelonDuskSeen";
+
+export const Rocket: React.FC = () => {
+  const { goblinService } = useContext(Context);
+  const [
+    {
+      context: { state },
+    },
+  ] = useActor(goblinService);
+  const [isDialogOpen, setIsDialogOpen] = useState(false);
+  const [isItemsOpen, setIsItemsOpen] = useState(false);
+  const [isRocketLaunching, setIsRocketLaunching] = useState(false);
+  const [isRocketLaunchComplete, setIsRocketLaunchComplete] = useState(false);
+  const [hasCompletedMission, setHasCompletedMission] = useState(false);
+  const [isMintingObservatory, setIsMintingObservatory] = useState(false);
+  const [observatoryMintError, setObservatoryMintError] =
```

Comment:
When a component has 4 different states it is usually a good indicator to split the functionality or use a state management solution like xstate.

To help maintain this complex state I would suggest introducing a separate business logic handler or state machine.